File: cpp_solutions/122. Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int ans = 0;
        
        int curr = prices[0];
        
        for (int x=0;x<prices.size();x++){
            
            if (prices[x] < curr){
                curr = prices[x];
            }
            else{
                ans+= prices[x] - curr;
                curr = prices[x];
            }
        }
        return ans;
        
    }
};

File: python3_solution_set0/122. Best Time to Buy and Sell Stock II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        
        ans = 0
        
        for x in range(1,len(prices)):
            if prices[x] > prices[x-1]:
                ans+=prices[x] - prices[x-1]
        return ans
            
            
        
